src/mod/codecs/mod_opus/opus_parse.c

```diff
@@ -0,0 +1,385 @@
+ * the Initial Developer. All Rights Reserved.
+ *
+ * Contributor(s):
+ *
+ * Claude Lamblin
+ * Julien Chavanton <jchavanton@gmail.com>
+ *
+ */
+
+#include "switch.h"
+#include <opus/opus.h>
+#include "opus_parse.h"
+/* Tables for LBRR_sympbol decoding */
+
+static const opus_int16 silk_LBRR_flags_2_PDFCum[3] = {53,106,256}; /* 256 - silk_LBRR_flags_2_iCDF[i] ; silk_LBRR_flags_2_iCDF[ 3 ] = { 203, 150, 0 };*/
+static const opus_int16 silk_LBRR_flags_3_PDFCum[7] = {41,61,90,131,146,174,256}; /* 256 - silk_LBRR_flags_3_iCDF[i] ; silk_LBRR_flags_3_iCDF[ 7 ] = { 215, 195, 166, 125, 110, 82, 0 };*/
+
+/* get the number of VAD flags - i.e. number of 20 ms frame - from the config */
+/* in a silk-only  or hybrid opus frame  mono or stereo*/
+/* 5 MSB TOC byte (see table 2 of IETF RFC6716  clause 3.1) */
+/* if 10 ms frame (config=0, 4, 8, 12, 14) : return 1 */
+/* if CELT_only frame no VAD flag =>return 0 */
+static opus_int16 switch_opus_get_nb_flags_in_silk_frame(int16_t config)
+{
+	opus_int16 silk_frame_nb_flags;
+	if(config >15) {
+		/* CELT_only frame no VAD flag nor LBRR flag */
+		silk_frame_nb_flags = 0;
+	}
+	else {
+		silk_frame_nb_flags = 1;  /*default*/
+		if (config < 12) {
+			/* silk-only NB, MB or WB */
+			/* The least two significant bits give the number of VAD flags inside the silk frame 1, 2 or 3 */
+			silk_frame_nb_flags  = config & 0x3;
+			if(silk_frame_nb_flags  == 0) { /* 0  => 10ms frame : one  VAD flag */
+				silk_frame_nb_flags++;
+			}
+		}
+	}
+	return(silk_frame_nb_flags);
+}
+
+/* get the time in ms corresponding to one VAD flag from the config */
+/* in a silk-only  or hybrid opus frame  mono or stereo*/
+/* 5 MSB TOC byte (see table 2 of IETF RFC6716  clause 3.1) */
+/* if CELT_only frame (config >15) no VAD flag =>return FALSE */
+/* if 10 ms frame (config=0, 4, 8, 12, 14) : return 10 */
+/* otherwise return 20 */
+static opus_int16 switch_opus_get_silk_frame_ms_per_flag(int16_t config, opus_int16 silk_frame_nb_flags)
+{
+	opus_int16 silk_size_frame_ms_per_flag;
+	if(config >15) {
+		/* CELT_only frame no VAD flag nor LBRR flag */
+		switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "switch_opus_get_silk_frame_ms_per_flag: code not written for CELT-only mode\n");
+		return FALSE;
+	}
+	silk_size_frame_ms_per_flag = 20;  /* default*/
+	if( silk_frame_nb_flags == 1) {    /* could be 10 or 20 ms */
+		if( (config &0x01) == 0) {
+			silk_size_frame_ms_per_flag  = 10;
+		}
+	}
+	return(silk_size_frame_ms_per_flag);
+}
+
+/* code written only for mono, silk-only or hybrid mode */
+/* for CELT-only frame no vad flags for LBRR flag the routine must not be called */
+/* for stereo : the mid frame VAD_flags and the LBRR_flag could be obtained */
+/* yet, to get the LBRR_flags of the mid frame the routine should be modified */
+/* to skip the side VAD flags and the side LBRR flag and to get the mid LBRR_symbol */
+static bool_t switch_opus_get_VAD_LBRR_flags(const uint8_t *buf, opus_int16 silk_frame_nb_flags, 
```

Comment:
why do we need to take the LBRR (FEC) flags here too ? we already have switch_opus_has_fec() .